Code/src/main/java/nl/utwente/group10/ui/components/lines/Connection.java

```diff
@@ -240,45 +242,83 @@ public void setStartAnchor(OutputAnchor start) {
             Number oldValue, Number newValue) {
-        updateStartEndPositions();
+        invalidateConnectionState();
     }
 
+    /**
+     * @return Whether or not both sides of this Connection are connected to an Anchor.
+     */
     public final boolean isConnected() {
         return startAnchor.isPresent() && endAnchor.isPresent();
     }
 
     /**
+     * Removes this Connection, disconnecting its anchors and removing this Connection from the pane it is on.
+     */
+    public final void remove() {
+        disconnect();
+        getPane().getChildren().remove(this);
+    }
+    
+    /**
      * Properly disconnects the given anchor from this Connection, notifying the anchor of its disconnection.
      */
     public final void disconnect(ConnectionAnchor anchor) {
         if (startAnchor.isPresent() && startAnchor.get().equals(anchor)) {
-            startAnchor.get().removeConnection(this);
             startAnchor = Optional.empty();
+            anchor.disconnectConnection(this);
+            ConnectionCreationManager.nextConnectionState();
         }
         if (endAnchor.isPresent() && endAnchor.get().equals(anchor)) {
-            endAnchor.get().removeConnection(this);
             endAnchor = Optional.empty();
+            anchor.disconnectConnection(this);
+            ConnectionCreationManager.nextConnectionState();
         }
-    }
 
-    public final void disconnect(Optional<? extends ConnectionAnchor> anchor) {
-        anchor.ifPresent(this::disconnect);
+        //Let the now (potentially) disconnected block update its visuals.
+        anchor.getBlock().invalidateConnectionState();
+        //Let the remaining connected anchors update their visuals.
+        invalidateConnectionStateCascading();
     }
-
+    
+    /**
+     * Disconnects both anchors.
+     */
     public final void disconnect() {
-        disconnect(startAnchor);
-        disconnect(endAnchor);
+        startAnchor.ifPresent(a -> disconnect(a));
+        endAnchor.ifPresent(a -> disconnect(a));
     }
 
     /**
      * Runs both the update Start end End position functions. Use when
      * refreshing UI representation of the Line.
      */
-    private void updateStartEndPositions() {
+    private void invalidateConnectionState() {
         startAnchor.ifPresent(a -> setStartPosition(a.getCenterInPane()));
         endAnchor.ifPresent(a -> setEndPosition(a.getCenterInPane()));
     }
 
+    /**
+     * Does the same as invalidateConnectionVisuals(), but cascading down to
+     * other blocks which are possibly also (indirectly) affected by the state
+     * change.
+     * 
+     * @param state
+     *            The newest visual state
+     */
+    public void invalidateConnectionStateCascading(int state) {
+        invalidateConnectionState();
+        startAnchor.ifPresent(a -> a.getBlock().invalidateConnectionStateCascading(state));
+        endAnchor.ifPresent(a -> a.getBlock().invalidateConnectionStateCascading(state));
```

Comment:
Ja dat klopt, maar dit wordt in Connection aangeroepen, naar Blocks. Blocks roepen vervolgens altijd naar beneden aan en nooit opnieuw op een Connection.

Dit heeft geloof ik inderdaad eerder meegespeelt met verwarring, maar dit is nu verholpen.
Dit creëerd nu geen loops.